### PR TITLE
Bump to `polkadot-v0.9.42` as a base for substrate related deps

### PR DESCRIPTION
**Pull request by dmitrylavrenov**

### CODE REVIEW

File: crates/humanode-peer/src/cli/run.rs
Comment (MOZGIII) [inline crates/humanode-peer/src/cli/run.rs:236]: We should unpack the `partial.other` to be more expressive.

File: crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs
Comment (MOZGIII) [inline crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs:63]: Do this with `tokio::block_in_place`

File: crates/humanode-peer/src/cli/run.rs
Comment (dmitrylavrenov) [reply crates/humanode-peer/src/cli/run.rs]: https://github.com/humanode-network/humanode/pull/861/commits/6694353680fde8312e12144c7dede29b8d0063b9

File: crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs
Comment (MOZGIII) [reply crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs:63]: Blocking I/O on reactor thread?

**Comment by MOZGIII** [comment]
This needs a rebase

File: crates/humanode-runtime/src/weights/mod.rs
Comment (MOZGIII) [inline crates/humanode-runtime/src/weights/mod.rs:1]: This is interesting, we probably should make sure this doesn't change in this PR.

File: crates/pallet-currency-swap/src/benchmarking.rs
Comment (MOZGIII) [inline crates/pallet-currency-swap/src/benchmarking.rs:49]: Isn't `Currency` deprecated?

File: crates/pallet-pot/src/lib.rs
Comment (MOZGIII) [inline crates/pallet-pot/src/lib.rs:217]:
```suggestion
impl<T: Config<I>, I: 'static> OnUnbalanced<CreditOf<T, I>> for Pallet<T, I> {
```

File: utils/e2e-tests/bash/fixtures/help.stdout.txt
Comment (MOZGIII) [inline utils/e2e-tests/bash/fixtures/help.stdout.txt:163]: This looks broken, we should see if there's a fix available

**Review by MOZGIII — COMMENTED** [review]
Went though the patch, looks fine - will review the commits to substrate/frontier upstreams again now, and we might be ready to merge once we sort the questions:

1. What to do with the `Currency` - ideally deprecate it from everywhere.
2. The comment I left about the new pot on_unbalanced implementation.

File: crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs
Comment (dmitrylavrenov) [reply crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs:63]: Discussed: we need to verify that keystore operations are used with either tokio::block_in_place or spawn::blocking (prefer option).

File: crates/pallet-pot/src/lib.rs
Comment (dmitrylavrenov) [reply crates/pallet-pot/src/lib.rs]: https://github.com/humanode-network/humanode/pull/861/commits/a8472217e441eaba2e9b434848739ded4516d30c

File: crates/pallet-pot/src/lib.rs
Comment (dmitrylavrenov) [reply crates/pallet-pot/src/lib.rs:76]: https://github.com/humanode-network/humanode/issues/1002

File: crates/pallet-pot/src/lib.rs
Comment (MOZGIII) [inline crates/pallet-pot/src/lib.rs:225]: Fix the clippy offense

File: utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh
Comment (MOZGIII) [inline utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh:11]: I'm thinking, based on https://github.com/paritytech/substrate/pull/13684, this should be
```suggestion
  --checks=all \
```

File: crates/pallet-pot/src/lib.rs
Comment (MOZGIII) [reply crates/pallet-pot/src/lib.rs]: Well, the also need to update the runtime config accordingly. Also, this breaks clippy.

File: utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh
Comment (dmitrylavrenov) [reply utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh:11]: It won't work as it includes `pre-migrate` and `post-migrate` checks. As an example, `pallet-humanode-session` will try to run `pre-migrate` on running network and check that there are no session identities, but `pallet-humanode-session` migration was done on live network and session identities list is not empty now.

File: crates/pallet-pot/src/lib.rs
Comment (dmitrylavrenov) [reply crates/pallet-pot/src/lib.rs]: Fixed

File: crates/pallet-pot/src/lib.rs
Comment (dmitrylavrenov) [reply crates/pallet-pot/src/lib.rs]: https://github.com/humanode-network/humanode/pull/861/commits/12704682582c189efb32875110c7962981892c73

File: crates/humanode-runtime/src/weights/mod.rs
Comment (dmitrylavrenov) [reply crates/humanode-runtime/src/weights/mod.rs]: https://github.com/humanode-network/humanode/pull/861/commits/e1177f01591ccdd5b9d469692d2241578dfb880a

File: utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh
Comment (MOZGIII) [reply utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh:11]: Let's have a call regarding this, but it seems like `pallet-humanode-session`s `pre-migrate` and `post-migrate` checks are just broken...

File: crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs
Comment (dmitrylavrenov) [reply crates/humanode-peer/src/cli/subcommand/bioauth/key/insert.rs:65]: https://github.com/humanode-network/humanode/pull/861/commits/29f2fdc71c20bc3b23b3b76a0d92e9aff2434f04

File: utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh
Comment (dmitrylavrenov) [reply utils/e2e-tests/bash/tests/try-runtime/on-runtime-upgrade-live.sh:11]: https://github.com/humanode-network/humanode/pull/1005

File: utils/e2e-tests/bash/fixtures/help.stdout.txt
Comment (dmitrylavrenov) [reply utils/e2e-tests/bash/fixtures/help.stdout.txt:163]: Just checked in `substrate: polkadot-v0.9.42`, the same help stdout.

In our `humanode` codebase we use the latest clap versions `4.5.4`.

File: crates/humanode-runtime/src/tests/claims_and_vesting.rs
Comment (MOZGIII) [inline crates/humanode-runtime/src/tests/claims_and_vesting.rs:435]: Not sure this is correct

File: crates/humanode-runtime/src/lib.rs
Comment (MOZGIII) [inline crates/humanode-runtime/src/lib.rs:459]: Those are incorrectly set for our chain

Comment (MOZGIII) [reply crates/humanode-runtime/src/lib.rs:459]: Do we need to rewrite vesting locks for this to work?

**Review by MOZGIII — CHANGES_REQUESTED** [review]
Runtime tests change in an odd way, once that it sorted we can start testing

File: crates/humanode-runtime/src/lib.rs
Comment (MOZGIII) [reply crates/humanode-runtime/src/lib.rs:459]: We need to test token claims, ideally with e2e tests

File: crates/author-ext-api/src/lib.rs
Comment (MOZGIII) [inline crates/author-ext-api/src/lib.rs:20]: This diff looks like it doesn't belong here

Comment (dmitrylavrenov) [reply crates/author-ext-api/src/lib.rs:20]: The changes have been applied after merging master containing updated `rust-toolchain`. Clippy noted an error  here before.

File: crates/pallet-vesting/src/api.rs
Comment (MOZGIII) [inline crates/pallet-vesting/src/api.rs:8]: This is something that only appeared after I last viewed the code - why this change?

File: crates/bioauth-flow-api/src/lib.rs
Comment (MOZGIII) [inline crates/bioauth-flow-api/src/lib.rs:26]: Same as https://github.com/humanode-network/humanode/pull/861/files#r1750787156

File: crates/pallet-vesting/src/api.rs
Comment (dmitrylavrenov) [reply crates/pallet-vesting/src/api.rs:8]: Just checked all commits after your last review. It wasn't added there. It was before.

File: crates/pallet-vesting/src/api.rs
Comment (MOZGIII) [inline crates/pallet-vesting/src/api.rs:18]: Same as https://github.com/humanode-network/humanode/pull/861/files#r1750787156

File: crates/pallet-vesting/src/api.rs
Comment (dmitrylavrenov) [reply crates/pallet-vesting/src/api.rs:8]: https://github.com/humanode-network/humanode/pull/861#discussion_r1750791085

File: crates/bioauth-flow-api/src/lib.rs
Comment (dmitrylavrenov) [reply crates/bioauth-flow-api/src/lib.rs:26]: https://github.com/humanode-network/humanode/pull/861#discussion_r1750791085

File: crates/pallet-vesting/src/api.rs
Comment (dmitrylavrenov) [reply crates/pallet-vesting/src/api.rs:18]: https://github.com/humanode-network/humanode/pull/861#discussion_r1750791085

File: crates/author-ext-api/src/lib.rs
Comment (dmitrylavrenov) [reply crates/author-ext-api/src/lib.rs:20]: image

File: utils/e2e-tests/bash/fixtures/help-output/help.benchmark.block.stdout.txt
Comment (dmitrylavrenov) [reply utils/e2e-tests/bash/fixtures/help-output/help.benchmark.block.stdout.txt:1]: Haven't checked future versions. Should we find the fix and backport it here ?

File: crates/author-ext-api/src/lib.rs
Comment (MOZGIII) [reply crates/author-ext-api/src/lib.rs:20]: Why is it in this diff? Should be fixed in `master`

File: crates/pallet-vesting/src/api.rs
Comment (MOZGIII) [reply crates/pallet-vesting/src/api.rs:8]: What was changed in this file then? Might be the ` where Id: Encode` change below?

File: crates/author-ext-api/src/lib.rs
Comment (dmitrylavrenov) [reply crates/author-ext-api/src/lib.rs:20]: Clippy doesn't note it in master.

File: crates/humanode-runtime/src/tests/claims_and_vesting.rs
Comment (dmitrylavrenov) [reply crates/humanode-runtime/src/tests/claims_and_vesting.rs:435]: Don't remember exactly the root cause. But it's connected with such changes - https://github.com/paritytech/substrate/commit/f8a1dd4c836d7d29382ba66fc2fce40fc1364546#diff-92414c2ef872b5f2a6af9fde64c13dc4e883a706d3d2da5817caffc6e4d8d0fcR137 containing

```
impl<Balance: Saturating + Copy + Ord> AccountData<Balance> {
	pub fn usable(&self) -> Balance {
		self.free.saturating_sub(self.frozen)
	}

	/// The total balance in this account including any that is reserved and ignoring any frozen.
	pub fn total(&self) -> Balance {
		self.free.saturating_add(self.reserved)
	}
}
```

`Alice` here is bootnode. But need time to remind.

File: crates/author-ext-api/src/lib.rs
Comment (MOZGIII) [reply crates/author-ext-api/src/lib.rs:20]: It must be a change in the macro then... But actually what clippy says here is that you can omit the `Encode` requirement as it is defined by the macro now

Comment (dmitrylavrenov) [reply crates/author-ext-api/src/lib.rs:20]: Yep, but the interesting fact that if I omit the `Encode` that I have build error )

Comment (MOZGIII) [reply crates/author-ext-api/src/lib.rs:20]: Why did it only appear with the last rebase though?

File: crates/humanode-runtime/src/tests/claims_and_vesting.rs
Comment (MOZGIII) [reply crates/humanode-runtime/src/tests/claims_and_vesting.rs:435]: Well, this is clearly a change that breaks things - and the question is if this breakage is intended / acceptable; imagine what the consequences of this are.

I think this might be an exact famous case where test found an issue and was "fixed" instead of leading to the actual issue being fixed.
This needs a very serious justification, with proofs in both the intent, the upstream code and upstream test corrections, as well as the implication and cause analysis.

File: crates/humanode-runtime/src/tests/claims_and_vesting.rs
Comment (MOZGIII) [inline crates/humanode-runtime/src/tests/claims_and_vesting.rs:430]:
```suggestion
        // Alice account can't be reaped, so existential deposit is untouchable.
```

**Comment by MOZGIII** [comment]
We'll need to backport https://github.com/paritytech/polkadot-sdk/pull/2073 into our fork to preserve UX.

---

After that, let's run the balances logic upgrade tests and if ok ship it (as usual starting from the internal testing chain)

File: utils/e2e-tests/bash/fixtures/help-output/help.benchmark.block.stdout.txt
Comment (MOZGIII) [reply utils/e2e-tests/bash/fixtures/help-output/help.benchmark.block.stdout.txt:1]: We've decided to backport the fix from the future

File: crates/humanode-runtime/src/lib.rs
Comment (MOZGIII) [reply crates/humanode-runtime/src/lib.rs:455]: Checked - all good

**Comment by MOZGIII** [comment]
This goes in after the scan result blob work

**Comment by MOZGIII** [comment]
Please rebase and prepare this for the merge

**Comment by dmitrylavrenov** [comment]
> Please rebase / update and prepare this for the merge

Properly rebased. Will prepare and check upgrade in playground to be ready for the merge.

**Review by MOZGIII — APPROVED** [review]

**Pull request merged**
